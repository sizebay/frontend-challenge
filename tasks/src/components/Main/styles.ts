import styled from 'styled-components';

export const Container = styled.main`
  display: flex;
  justify-content: center;
  align-items: center;

  width: 100%;
  height: 100%;
`;

export const Content = styled.section`
  display: flex;
  flex-direction: column;

  width: 100%;
  max-width: 800px;
  height: 650px;

  padding: 0 60px;
  background-color: var(--white);
  border-radius: 4px;
  box-shadow: var(--box-shadow) 0 3px 6px;
`;
