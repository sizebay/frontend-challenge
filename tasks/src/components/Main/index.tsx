import React from 'react';

import Header from '../Header';

import { Container, Content } from './styles';

const Main: React.FC = () => {
  return (
    <Container>
      <Content>
        <Header />
      </Content>
    </Container>
  );
};

export default Main;
