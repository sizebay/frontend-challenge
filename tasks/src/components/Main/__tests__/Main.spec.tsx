import { render } from '@testing-library/react';

import Main from '..';

describe('<Main/>', () => {
  it('should match snapshot', () => {
    const container = render(<Main />);

    expect(container).toMatchSnapshot();
  });
});
