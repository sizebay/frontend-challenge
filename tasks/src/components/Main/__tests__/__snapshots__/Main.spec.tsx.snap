// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main/> should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  margin-top: 54px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 > strong {
  font-size: 60px;
  font-weight: 500;
}

.c3 > div > strong {
  font-size: 20px;
  font-weight: 400;
}

.c3 > div > p {
  font-size: 20px;
  font-weight: 300;
  margin: 0;
}

.c4 {
  font-size: 24px;
  font-weight: 400;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  max-width: 800px;
  height: 650px;
  padding: 0 60px;
  background-color: var(--white);
  border-radius: 4px;
  box-shadow: var(--box-shadow) 0 3px 6px;
}

<body>
    <div>
      <main
        class="c0"
      >
        <section
          class="c1"
        >
          <header
            class="c2"
          >
            <h3
              class="c3"
            >
              <strong>
                10
              </strong>
              <div>
                <strong>
                  Sep
                </strong>
                <p>
                  2021
                </p>
              </div>
            </h3>
            <h4
              class="c4"
            >
              Friday
            </h4>
          </header>
        </section>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="sc-hKgJUU jdMTkQ"
    >
      <section
        class="sc-eCstlR jHxlKA"
      >
        <header
          class="sc-bdfBQB eOzPlJ"
        >
          <h3
            class="sc-gsTEea UvZWt"
          >
            <strong>
              10
            </strong>
            <div>
              <strong>
                Sep
              </strong>
              <p>
                2021
              </p>
            </div>
          </h3>
          <h4
            class="sc-dlfnuX gfXCJR"
          >
            Friday
          </h4>
        </header>
      </section>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
